packages/components/autocomplete/src/use-autocomplete.ts

```diff
@@ -203,6 +204,8 @@ export function useAutocomplete<T extends object>(originalProps: UseAutocomplete
   const inputRef = useDOMRef<HTMLInputElement>(ref);
   const scrollShadowRef = useDOMRef<HTMLElement>(scrollRefProp);
 
+  const [shouldFocus, setShouldFocus] = useState(false);
```

Comment:
@wingkwong could you try using a `useRef` instead? so we avoid re-rendering the component